llvm/lib/Target/X86/X86ISelLowering.cpp

```diff
@@ -1092,6 +1092,8 @@ X86TargetLowering::X86TargetLowering(const X86TargetMachine &TM,
     setOperationAction(ISD::FABS,               MVT::v2f64, Custom);
     setOperationAction(ISD::FCOPYSIGN,          MVT::v2f64, Custom);
 
+    setOperationAction(ISD::LRINT, MVT::v4f32, Custom);
```

Comment:
Is this correct? The MVT of `ISD::LRINT` is usually set to its output type, not input type. Shouldn't this be `MVT::v4i32`?

Reply:
Yes, it's correct. `ISD::LRINT` uses input type, see https://github.com/llvm/llvm-project/blob/main/llvm/lib/CodeGen/SelectionDAG/LegalizeDAG.cpp#L1002-L1006